### PR TITLE
Don't retry 500 responses

### PR DESCRIPTION
**Pull request by macobo**
## Changes

Retrying won't do any good here, it just causes spam. In this case a needless alert.

## Checklist
- [ ] Tests for new code (if applicable)
- [ ] TypeScript definitions (module.d.ts) updated and in sync with library exports (if applicable)

### CODE REVIEW

**Review by fuziontech — APPROVED** [review]
🚢

**Comment by macobo** [comment]
Gonna wait for a review from yakko - shipping this won't change the ongoing issue.

Also worth thinking about - are we infinitely retrying? Exponential backoff is really important if so.

**Review by yakkomajuri — APPROVED** [review]
I'm ok with this.

But I guess the original rationale was that client-side errors are a lost cause, since the payload has an issue so retrying it won't fix it. With 5xx errors on the other hand, you could argue that the server could potentially recover before subsequent retries (automatically or with human input). Think maxed out DB connections, or a bug we quickly fix.

**Pull request merged**
